_search-plugins/ubi/data-structures.md

```diff
@@ -0,0 +1,205 @@
+---
+layout: default
+title: UBI client data structures
+parent: User behavior insights
+has_children: false
+nav_order: 7
+---
+
+# Sample client data structures
+
+The following client data structures can be used to create events that follow the [UBI event schema specification](https://github.com/o19s/ubi).
+For more information about the schema, see [UBI index schemas]({{site.url}}{{site.baseurl}}/search-plugins/ubi/schemas/).
+
+
+You must provide an implementation for the following functions:
+- `getClientId()`
+- `getQueryId()`
+  
+You can also optionally provide an implementation for the following functions:
+- `getSessionId()`
+- `getPageId()`
+
+<!-- Not needed with this page: Other sample implementations can be found [here](#TODO-clients-link). -->
+
+These **JavaScript** structures below give you a starter implementation that serializes into JSON.
```

Comment:
Serialize what into JSON?

Reply:
Global: 
- Add a new line between an intro sentence and a code block.
- Use "following" instead of "below"
- Introduce code blocks with a sentence ending in a colon, not a period.